src/Servers/IIS/IntegrationTesting.IIS/src/IISDeployer.cs

```diff
@@ -473,6 +490,18 @@ private static void RetryServerManagerAction(Action<ServerManager> action)
             delay *= 1.5;
         }
 
+        // Try to upload the applicationHost config on helix to help debug
+        UploadFileOnHelix(_applicationHostConfig, "serverManagerRetryFailed.applicationHost.config");
+
         throw new AggregateException($"Operation did not succeed after {retryCount} retries", exceptions.ToArray());
     }
+
+    private static void UploadFileOnHelix(string filePath, string uploadFileName)
+    {
+        var HELIX_WORKITEM_UPLOAD_ROOT = Environment.GetEnvironmentVariable("HELIX_WORKITEM_UPLOAD_ROOT");
+        if (!string.IsNullOrEmpty(HELIX_WORKITEM_UPLOAD_ROOT))
+        {
+            File.Copy(filePath, Path.Combine(HELIX_WORKITEM_UPLOAD_ROOT, uploadFileName));
```

Comment:
```suggestion
            File.Copy(filePath, Path.Combine(HELIX_WORKITEM_UPLOAD_ROOT, uploadFileName, overwrite: true));
```

This may be called multiple times as it's used in the `RetryServerManagerAction` lambda